cmd/download.go

```diff
@@ -47,9 +55,22 @@ func runFetch(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
+
+	// Ensure ARTIFACTORY_TOKEN is set if cache URLs require it (new part starts here)
+	token := os.Getenv("ARTIFACTORY_TOKEN")
+	if token == "" {
```

Comment:
I don't think that you *always* want to warn if `ARTIFACTORY_TOKEN` is not set -- you only want to *error* if you *need* the Artifactory token and it's not set (i.e., if you have a cache that you need to download from).